src/passes/pass.cpp

```diff
@@ -495,6 +495,9 @@ void PassRegistry::registerPasses() {
   registerPass("type-unfinalizing",
                "mark all types as non-final (open)",
                createTypeUnFinalizingPass);
+  registerPass("unsubtyping",
+               "removes unnecessary subtyping relationships",
+               createUnsubtypingPass);
```

Comment:
Should we run this in the default pipeline?

Reply:
Maybe, but I'm worried that the cast analysis is going to make this extremely expensive. It would be nice to get some data on that first.

Reply:
The pass runs in 0.8 seconds on optimized calc.wasm after running TypeSSA or 0.6 seconds without TypeSSA, so I think it is fast enough that we could put it in the default pipeline.

Reply:
Great!

Let's leave the default pipeline for a followup, unless we are certain this has no downsides? Intuitively I think this can only improve optimizability, but there are theoretical cases like it preventing a merging of similar functions (if the merge depending on subtyping; we don't do that atm, but it is one idea we've had).

Reply:
This pass still runs in 12 seconds and cannot optimize anything on the unoptimized binary. After the first -O3, it takes 7.5 seconds (and I don't know if it optimizes anything there).

Reply:
12 is maybe a lot. An option might be to add a limit on the number of iterations and do a conservative estimate from there, maybe.

Reply:
Another option might be to only run this in `-O3` and such, and recommend the J2Wasm first pass be `-O1` or `-O2` only, which might make sense actually anyhow.

Reply:
Happy to experiment (or have j2wasm experiment) with different pass placements after we get this landed. It would be interesting to see what optimizations unblock this one from working.